jobs/job.go

```diff
@@ -174,7 +174,7 @@ type Config interface {
 // Manager is a manager for centrifuge Jobs.
 type Manager interface {
 	// ExecuteWithinJob executes the given unit of work within a Job
-	ExecuteWithinJob(ctx context.Context, accountID identity.DID, existingJobID JobID, desc string, work func(accountID identity.DID, jobID JobID, jobManager Manager, err chan<- error)) (jobID JobID, done chan bool, err error)
+	ExecuteWithinJob(ctx context.Context, accountID identity.DID, existingJobID JobID, desc string, work func(accountID identity.DID, jobID JobID, jobManager Manager, err chan<- error)) (jobID JobID, done chan error, err error)
```

Comment:
could you check the if the `err chan<- error` is needed here anymore? I think we only need one.

Reply:
I believe we still need the inner job error channel. since it tells asynchronously when the actual work is done. Afterwards the wrapper job logic needs a different channel to publish the result of it, since the `ExecuteWithinJob` adds more logic on top of the `go work()` step